nomad/csi_endpoint.go

```diff
@@ -471,6 +482,7 @@ func (v *CSIVolume) controllerPublishVolume(req *structs.CSIVolumeClaimRequest,
 		ClientCSINodeID: externalNodeID,
 		AttachmentMode:  req.AttachmentMode,
 		AccessMode:      req.AccessMode,
+		MountOptions:    csiVolumeMountOptions(vol.MountOptions),
```

Comment:
Leaving a note for myself here: this confused the heck out of me, but it looks like we crossed some wires between `client/structs` and `nomad/structs` where we shouldn't have. I need to check what that looks like on the wire to see if we can fix it without breaking backwards compat.